### PR TITLE
kmymoney: 5.1.1 -> 5.1.2

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/148779

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/kmymoney/default.nix [hunk 1]
```diff
@@ -10,48 +17,36 @@
...
-    patchPythonScript "$out/share/kmymoney/weboob/kmymoneyweboob.py"
-
-    # Within the embedded Python interpreter, sys.argv is unavailable, so let's
-    # assign it to a dummy value so that the assignment of sys.argv[0] injected
-    # by patchPythonScript doesn't fail:
-    sed -i -e '1i import sys; sys.argv = [""]' \
-      "$out/share/kmymoney/weboob/kmymoneyweboob.py"
+  preFixup = ''
+    qtWrapperArgs+=(
+      --prefix PYTHONPATH : "${python}/${python.sitePackages}"
```
Comment (dotlambda) [inline pkgs/applications/office/kmymoney/default.nix:48]: This doesn't work; KMyMoney doesn't seem to respect `$PYTHONPATH`.

Comment (dotlambda) [reply pkgs/applications/office/kmymoney/default.nix:48]: Does someone know enough about the interactions between C++ and Python to find out how to set `$PYTHONPATH` in https://github.com/KDE/kmymoney/blob/v5.1.2/kmymoney/plugins/weboob/interface/weboobinterface.cpp?

File: pkgs/applications/office/kmymoney/pythonpath.patch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+diff --git a/kmymoney/plugins/weboob/interface/weboobinterface.cpp b/kmymoney/plugins/weboob/interface/weboobinterface.cpp
+index 9d8ad440d..98de63df9 100644
+--- a/kmymoney/plugins/weboob/interface/weboobinterface.cpp
++++ b/kmymoney/plugins/weboob/interface/weboobinterface.cpp
+@@ -65,6 +65,8 @@ WeboobInterface::WeboobInterface() :
+ 
+         PyObject *sys = PyImport_ImportModule("sys");
+         PyObject *path = PyObject_GetAttrString(sys, "path");
++        PyObject *pythonpath = PyUnicode_FromString("@pythonpath@");
++        PyList_Append(path, pythonpath);
```
Comment (dotlambda) [inline pkgs/applications/office/kmymoney/pythonpath.patch:10]: I really wonder why this doesn't work either.

**Comment by mweinelt** [comment]
Sorry, not my cup of tea.

**Pull request closed without merging**